server/src/com/thoughtworks/go/config/update/TemplateConfigCommand.java

```diff
@@ -46,7 +46,7 @@ public TemplateConfigCommand(PipelineTemplateConfig templateConfig, LocalizedOpe
     public boolean isValid(CruiseConfig preprocessedConfig) {
         TemplatesConfig templatesConfig = preprocessedConfig.getTemplates();
         preprocessedTemplateConfig = findAddedTemplate(preprocessedConfig);
-        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(preprocessedConfig));
+        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(templatesConfig));
```

Comment:
Because of https://github.com/gocd/gocd/blob/master/config/config-api/src/com/thoughtworks/go/config/FetchTask.java#L234 ?
We do seem to be validating the task once the template is used in a pipeline and that's what needs to be done because a fetch artifact task needs a upstream pipeline material that can only be configured for a pipeline and not template?